drivers/clock_control/nrf_clock_calibration.c

```diff
@@ -215,6 +212,15 @@ static void measure_temperature(struct k_work *work)
 			started ? "started" : "skipped", diff);
 }
 
+#if DT_HAS_NODE(DT_NODELABEL(temp))
+static inline struct device *temp_device(void)
+{
+	return device_get_binding(DT_LABEL(DT_NODELABEL(temp)));
```

Comment:
Does the `temp` device need to be NODELABEL based?  Seems like DT_INST would be fine in this case.  We should only use NODELABEL in the cases that its really required.

Reply:
> Does the `temp` device need to be NODELABEL based?

Yes, that is my interpretation: https://github.com/zephyrproject-rtos/zephyr/pull/23744#discussion_r397745301

Reply:
I see 2 ways a device should or shouldn't exist:
1. There is no node w/compat for the device in the dt
2. there is a node w/compat and `status = disabled`

Reply:
In the thread @anangl even gives the snippet using `DT_INST` form.